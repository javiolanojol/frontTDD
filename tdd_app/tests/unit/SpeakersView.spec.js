import { shallowMount,createLocalVue } from '@vue/test-utils'
import SpeakersView from '@/views/SpeakersView'
import SearchForm from '@/components/SearchForm'
import SpeakersList from '@/components/SpeakersList'
import Vuex from 'vuex'

const localVue = createLocalVue()

localVue.use(Vuex)

let speaker = {
    name:'My name'
}

describe('SpeakersView', () => {

    let store, getters

    beforeEach(() => {

        getters = {
            speakers: () => [speaker, speaker]
        }
        store = new Vuex.Store({
            getters
        })
    })

    const build = () =>{
        const wrapper = shallowMount(SpeakersView,{
            store,
            localVue
        })

        return{
            wrapper,
            SpeakersList: () => wrapper.find(SpeakersList),
            SearchForm: () => wrapper.find(SearchForm)
        }

    }

    it('it should render components', () => {

        const {wrapper} = build()

        expect(wrapper.html()).toMatchSnapshot()
    })

    it('it should render the right components', () => {

        const{SpeakersList,SearchForm} = build();

        expect(SpeakersList().exists()).toBe(true)
        expect(SearchForm().exists()).toBe(true)
    })

    it('should pass speakers to the SpeakersList', () =>{

        const{SpeakersList} = build();

        expect(SpeakersList().vm.speakers).toEqual([speaker,speaker])
        expect(SpeakersList().vm.speakers).toBe(store.getters.speakers)

    })
})