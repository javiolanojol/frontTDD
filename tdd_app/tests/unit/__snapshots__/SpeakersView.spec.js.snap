// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpeakersView it should render components 1`] = `
<div>
  <searchform-stub></searchform-stub>
  <speakerslist-stub speakers="[object Object],[object Object]"></speakerslist-stub>
</div>
`;
